dashboard/test/models/teacher_feedback_test.rb

```diff
@@ -62,4 +62,23 @@ class TeacherFeedbackTest < ActiveSupport::TestCase
     assert_equal(feedbacks[0], TeacherFeedback.where(student: students[0]).latest)
     assert_equal(feedbacks[1], TeacherFeedback.where(student: students[1]).latest)
   end
+
+  test 'destroys when teacher is destroyed' do
+    teacher = create :teacher
+    first_feedback = create :teacher_feedback, teacher: teacher
+    second_feedback = create :teacher_feedback, teacher: teacher
+    teacher.destroy
+    assert_not TeacherFeedback.exists?(first_feedback.id)
+    assert_not TeacherFeedback.exists?(second_feedback.id)
+  end
+
+  test 'does not destroy when student is destroyed' do
+    teacher = create :teacher
+    student = create :student
+    first_feedback = create :teacher_feedback, teacher: teacher, student: student
+    second_feedback = create :teacher_feedback, teacher: teacher, student: student
+    student.destroy
+    assert TeacherFeedback.exists?(first_feedback.id)
+    assert TeacherFeedback.exists?(second_feedback.id)
```

Comment:
If you wanted to test that the soft-deleted feedback can be recovered with the user, I believe you'd call
```rb
user.restore(:recursive => true, :recovery_window => 2.minutes)
```
and then verify that the feedback had been restored.

Reply:
Or call User.undestroy: https://github.com/code-dot-org/code-dot-org/blob/66b8da828d0bc929ff7b13a29d0ea75ac3d402fb/dashboard/app/models/user.rb#L1958